Mathlib/AlgebraicTopology/SimplicialSet/Path.lean

```diff
@@ -85,4 +85,47 @@ lemma Path.ext' {n : ℕ} {f g : Path X (n + 1)}
       rw [← f.arrow_tgt (Fin.last n), ← g.arrow_tgt (Fin.last n), h]
   · exact h j
 
+/-- Maps of simplicial sets induce maps of paths in a simplicial set.-/
+@[simps]
+def mapPath {X Y : SSet.{u}} {n : ℕ} (σ : X ⟶ Y) (f : X.Path n) : Y.Path n where
+  vertex i := σ.app (Opposite.op [0]) (f.vertex i)
+  arrow i := σ.app (Opposite.op [1]) (f.arrow i)
+  arrow_src i := by
+    simp only [← f.arrow_src i]
+    exact (congr (σ.naturality (δ 1).op) (by rfl)).symm
+  arrow_tgt i := by
+    simp only [← f.arrow_tgt i]
+    exact (congr (σ.naturality (δ 0).op) (by rfl)).symm
+
+/-- `mapPath` respects subintervals of paths.-/
+lemma mapPath_interval {X Y : SSet.{u}} {n : ℕ} (σ : X ⟶ Y) (f : X.Path n)
+    (j l : ℕ) (hjl : j + l ≤ n) :
+    (mapPath σ f).interval j l hjl = mapPath σ (f.interval j l hjl) := rfl
+
+/-- The spine of the unique non-degenerate `n`-simplex in `Δ[n]`.-/
+def idSpine (n : ℕ) : Path Δ[n] n := spine Δ[n] n (standardSimplex.idSimplex n)
+
+/-- Any inner horn contains the spine of the unique non-degenerate `n`-simplex
+in `Δ[n]`.-/
+@[simps]
+def spineHorn {n : ℕ} (i : Fin (n + 3))
```

Comment:
This may be moved to the `horn` namespace, and you may add the following lemma:
```lean
@[simp]
lemma horn.spine_map_hornInclusion {n : ℕ} (i : Fin (n + 3))
    (h₀ : 0 < i) (hₙ : i < Fin.last (n + 2)) :
    mapPath (hornInclusion (n + 2) i) (spineHorn i h₀ hₙ) = idSpine (n + 2) := rfl
```